libraries/backend/azure_connection.rb

```diff
@@ -123,6 +117,12 @@ def rest_get_call(url, params = {})
   # https://docs.microsoft.com/en-us/rest/api/azure/
   #
   def authenticate(resource)
+    # Validate the presence of credentials.
+    unless @credentials.values.compact.delete_if(&:empty?).size == 4
+      raise HTTPClientError::MissingCredentials, 'The following must be set in the Environment:'\
+        " #{@credentials.keys}.\n"\
+        "Provided: #{@credentials}"
```

Comment:
This looks like this will currently output the key and value. Debating whether it's better to avoid outputting the values and just output which keys they have set.

Reply:
Can do, good idea.

Reply:
This should be fixed now.